src/tls/test/main.cpp

```diff
@@ -22,67 +22,205 @@ static const string contents_ =
 
 void corrupt(std::vector<uint8_t>& buf)
 {
+  buf[1]++;
   buf[buf.size() / 2]++;
+  buf[buf.size() - 2]++;
 }
 
+static constexpr tls::CurveImpl supported_curves[] = {
+  tls::CurveImpl::secp384r1,
+  tls::CurveImpl::curve25519,
+  tls::CurveImpl::secp256k1_mbedtls,
+  tls::CurveImpl::secp256k1_bitcoin};
+
+static constexpr char const* labels[] = {
```

Comment:
I guess this could be moved to `keypair.h`?

Reply:
To get labels like this we need to do the ugly `static_cast` from the `enum class` so I'd prefer to keep it local to the tests, where its only used for debug information. If we need labels elsewhere, we can add a more explicit solution to keypair.h.